### PR TITLE
[#267]_Module version in module.xml is missing when creating new module

### PR DESCRIPTION
**Pull request by dylanngo95**
#267 When creating a new module It missing setup_version in a module.xml file.
1, Fix bug
2, Update unit test
Please review and merge the code. 
Thanks

### CODE REVIEW

**Review by VitaliyBoyko — CHANGES_REQUESTED** [review]
Hi @dylanngo95 

QA failed 🚫 

* Specify Magento version 2.3.0+
image
* Create a module

*Actual result:*
setup_version is still present
image

*Expected result:*
setup_version should be applicable only for Magento <2.2.11

Additional issue:
Module name inspection triggered under setup_module:
image

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModuleDialog.java [hunk 1]
```diff
@@ -72,6 +73,8 @@
     private JLabel moduleDescriptionLabel;//NOPMD
     private JLabel moduleNameLabel;//NOPMD
 
+    private static final String MAGENTO_OLD_VERSION = "2.2.11";
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModuleDialog.java:76]: Let it be `MAGENTO_BEFORE_DECLARATIVE_SCHEMA_VERSION`. What do you think?

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModuleDialog.java [hunk 2]
```diff
@@ -224,10 +228,39 @@ public String getModuleDescription() {
...
+     * get Module Version.
+     *
+     * @return string
+     */
     public String getModuleVersion() {
         return this.moduleVersion.getText().trim();
     }
 
+    /**
+     * get Module Version.
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModuleDialog.java:241]: Please use capital lettering

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModuleDialog.java [hunk 3]
```diff
@@ -224,10 +228,39 @@ public String getModuleDescription() {
...
+     *
+     * @return string|null
+     */
+    public String getSetupVersion() {
+        String setupVersion = null;
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModuleDialog.java:246]: Unnecessary variable.

File: src/com/magento/idea/magento2plugin/generation/php/MagentoModuleGenerator.java [hunk 4]
```diff
@@ -25,61 +25,95 @@
...
 
+    @Override
     public boolean isPrimaryGenerator() {
         return PlatformUtils.isPhpStorm();
     }
 
-    public void generateProject(@NotNull Project project, @NotNull VirtualFile baseDir, @NotNull MagentoProjectGeneratorSettings settings, @NotNull Module module) {
-        Settings dataService = Settings.getInstance(project);
+    /**
+     * generateProject.
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/generation/php/MagentoModuleGenerator.java:67]: Use capital lettering

File: src/com/magento/idea/magento2plugin/generation/php/MagentoModuleGenerator.java [hunk 5]
```diff
@@ -25,61 +25,95 @@
...
+                        .getInstance()
+                        .refreshIncludePaths(dataService.getState(), project);
             });
         };
         StartupManager.getInstance(project).runWhenProjectIsInitialized(generate);
     }
 
-    private PsiFile generateComposerJson(@NotNull Project project, @NotNull PsiDirectory baseDir, @NotNull MagentoProjectGeneratorSettings settings) {
+    /**
+     * generateComposerJson.
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/generation/php/MagentoModuleGenerator.java:105]: Capital lettering

File: src/com/magento/idea/magento2plugin/generation/php/MagentoModuleGenerator.java [hunk 6]
```diff
@@ -93,7 +127,18 @@ private PsiFile generateComposerJson(@NotNull Project project, @NotNull PsiDirec
         ), project).generate(ACTION_NAME);
     }
 
-    private PsiFile generateRegistrationPhp(@NotNull Project project, @NotNull PsiDirectory baseDir, @NotNull MagentoProjectGeneratorSettings settings) {
+    /**
+     * generateRegistrationPhp.
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/generation/php/MagentoModuleGenerator.java:131]: Capital lettering

File: src/com/magento/idea/magento2plugin/util/magento/MagentoVersion.java [hunk 7]
```diff
@@ -70,4 +70,26 @@ public static String get(final Project project, final String magentoPath) {
     private static String getFilePath(final String magentoPath) {
         return magentoPath + File.separator + ComposerJson.FILE_NAME;
     }
+
+    /**
+     * compare Magento Version.
+     *
+     * @param str1 String
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/util/magento/MagentoVersion.java:77]: Please use self-explanatory variable names. `str1 ` doesn’t say anything to us. Also, please describe a few words about how this method works. It about to be commonly used. Thank you!

File: testData/actions/generation/generator/ModuleXmlGenerator/generateModuleFile/module.xml [hunk 8]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
-    <module name="Test_Module"/>
+    <module name="Test_Module" setup_version="1.0.0"/>
```
Comment (VitaliyBoyko) [inline testData/actions/generation/generator/ModuleXmlGenerator/generateModuleFile/module.xml:4]: Please leave these tests as they are, just add a new version config. For the case with `setup_version` we need a new test. Both cases should be covered though. Thanks!

**Review by VitaliyBoyko — CHANGES_REQUESTED** [review]

File: src/com/magento/idea/magento2plugin/generation/php/MagentoModuleGenerator.java (hunk 6 above)
Comment (VitaliyBoyko) [reply src/com/magento/idea/magento2plugin/generation/php/MagentoModuleGenerator.java:131]: Please check lettering everywhere.

**Comment by VitaliyBoyko** [comment]
Also please pull changes before making your ones. You don't need to force push.

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModuleDialog.java (hunk 1 above)
Comment (dylanngo95) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/NewModuleDialog.java:76]: Yep, It's better.

**Review by VitaliyBoyko — APPROVED** [review]

**Pull request merged**